controllers/k8ssandra/secrets.go

```diff
@@ -81,6 +87,11 @@ func (r *K8ssandraClusterReconciler) reconcileReaperSecrets(ctx context.Context,
 				return result.Error(err)
 			}
 			logger.Info("Reaper user secrets successfully reconciled")
+		} else if kc.Spec.IsAuthEnabled() && kc.Spec.Reaper.UseExternalSecrets() {
```

Comment:
Same here, kc.Spec.Reaper.UseExternalSecrets() should always be false I guess.
As long as we're in the k8ssandra-cluster controller, we should use `kc.Spec.UseExternalSecrets()` instead.